pyomo/contrib/piecewise/transform/disagreggated_logarithmic.py

```diff
@@ -0,0 +1,178 @@
+        transBlock.substitute_var.setlb(self.substitute_var_lb)
+        transBlock.substitute_var.setub(self.substitute_var_ub)
+
+        log_dimension = ceil(log2(num_simplices))
+        transBlock.log_simplex_indices = RangeSet(0, log_dimension - 1)
+        transBlock.binaries = Var(transBlock.log_simplex_indices, domain=Binary)
+
+        # Injective function B: \mathcal{P} -> {0,1}^ceil(log_2(|P|)) used to identify simplices
+        # (really just polytopes are required) with binary vectors. Any injective function
+        # is enough here.
+        B = {}
+        for i in transBlock.simplex_indices:
+            # map index(P) -> corresponding vector in {0, 1}^n
+            B[i] = self._get_binary_vector(i, log_dimension)
+
+        # The lambda variables \lambda_{P,v} are indexed by the simplex and the point in it
+        transBlock.lambdas = Var(
+            transBlock.simplex_indices, transBlock.simplex_point_indices, bounds=(0, 1)
+        )
+
+        # Numbered citations are from Vielma et al 2010, Mixed-Integer Models
+        # for Nonseparable Piecewise-Linear Optimization
+
+        # Sum of all lambdas is one (6b)
+        transBlock.convex_combo = Constraint(
+            expr=sum(
+                transBlock.lambdas[P, v]
+                for P in transBlock.simplex_indices
+                for v in transBlock.simplex_point_indices
+            )
+            == 1
+        )
+
+        # The branching rules, establishing using the binaries that only one simplex's lambdas
+        # may be nonzero
+        @transBlock.Constraint(transBlock.log_simplex_indices)  # (6c.1)
+        def simplex_choice_1(b, l):
+            return (
+                sum(
+                    transBlock.lambdas[P, v]
+                    for P in self._P_plus(B, l, transBlock.simplex_indices)
+                    for v in transBlock.simplex_point_indices
+                )
+                <= transBlock.binaries[l]
+            )
+
+        @transBlock.Constraint(transBlock.log_simplex_indices)  # (6c.2)
+        def simplex_choice_2(b, l):
+            return (
+                sum(
+                    transBlock.lambdas[P, v]
+                    for P in self._P_0(B, l, transBlock.simplex_indices)
+                    for v in transBlock.simplex_point_indices
+                )
+                <= 1 - transBlock.binaries[l]
+            )
+
+        # for i, (simplex, pwlf) in enumerate(choices):
+        # x_i = sum(lambda_P,v v_i, P in polytopes, v in V(P))
+        @transBlock.Constraint(transBlock.dimension_indices)  # (6a.1)
+        def x_constraint(b, i):
+            return pw_expr.args[i] == sum(
+                transBlock.lambdas[P, v]
+                * pw_linear_func._points[self.idx_to_simplex[P][v]][i]
+                for P in transBlock.simplex_indices
+                for v in transBlock.simplex_point_indices
+            )
+
+        # Make the substitute Var equal the PWLE (6a.2)
```

Comment:
6a.2 from the paper?

Reply:
Yes, all my comments with numbers are referencing a corresponding equation in the Vielma paper